src/main/java/com/openshift3/internal/client/DefaultClient.java

```diff
@@ -137,6 +137,23 @@ private IHttpClient newIHttpClient(ISSLCertificateCallback sslCertCallback){
 	}
 
 	@Override
+	public <T extends IResource> T update(T resource) {
+		try {
+			final URL endpoint = new URLBuilder(getBaseURL(), getTypeMappings())
+				.resource(resource)
+				.addParmeter("namespace", resource.getNamespace())
+				.build();
+			String response = client.put(endpoint, IHttpClient.DEFAULT_READ_TIMEOUT, resource);
+			LOGGER.debug(response);
+			return factory.create(response);
+		} catch (HttpClientException e){
+			throw handleHttpClientException("Exception creating the resource", e);
```

Comment:
@tomason Update the error message(s) to reflect the operation (e.g. creating->update)